main/entry.c

```diff
@@ -1021,7 +1021,14 @@ static void writeTagEntry (const tagEntryInfo *const tag)
 	else if (Option.etags)
 		length = writeEtagsEntry (tag);
 	else
+	{
+		/* This assertion should be at initTagEntry.
+		   However, ada parser does delayed initialization;
+		   it passes NULL pointer as a name to initTagEntry; and
+		   builds the name later in parsing. */
+		Assert (strchr (tag->name, '\t') == NULL);
```

Comment:
Maybe duplicate this in `initTagEntry()`, like `Assert (tag->name == NULL || strchr (tag->name, '\t') == NULL)`?  It would be a lot handier for everyone (but the Ada parser), especially as now `Assert()` reports input file position.